src/effects/backends/builtin/pitchshifteffect.cpp

```diff
@@ -79,15 +123,23 @@ void PitchShiftEffect::processChannel(
     Q_UNUSED(groupFeatures);
     Q_UNUSED(enableState);
 
-    const double pitchParameter = m_pPitchParameter->value();
+    if (m_currentFormant != m_pFormantPreservingParameter->toBool()) {
+        m_currentFormant = !m_currentFormant;
```

Comment:
Yeah, good idea. Anyway, didn't you mean this solution instead?

```
const bool formantPreserving = m_pFormantPreservingParameter->toBool();

if (m_currentFormant != formantPreserving) {
        m_currentFormant = formantPreserving;
```

Reply:
Both works. C++17 added support for `init-statement`s before the `condition`. 
https://en.cppreference.com/w/cpp/language/if
This will bind the variable scope to the if statement instead of the outer scope. I like to reduce the scope variables are bound to to be as narrow as possible.
If you find your solution easier to read/understand then go ahead and use that instead. As I said, both is legal C++

Reply:
> Both works. C++17 added support for `init-statement`s before the `condition`. https://en.cppreference.com/w/cpp/language/if https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2016/p0305r0.html 

Oh perfect, sounds cool. The new thing for me. Thank you for the links.

> This will bind the variable scope to the if statement instead of the outer scope. I like to reduce the scope variables are bound to to be as narrow as possible.

That's reasonable.

> If you find your solution easier to read/understand then go ahead and use that instead. As I said, both is legal C++

Reducing the variable scope makes sense to me.